### DISCUSSION TITLE
My planned setup with an Arducam IMX462 (B0423)

### DISCUSSION THREAD

**Discussion by Vertikar** [aaronwmorris/indi-allsky#935] [General]
Wanted to start this as I've been trying to get decent results out of the IMX462 with @aaronwmorris in this issue - https://github.com/aaronwmorris/indi-allsky/issues/878

But wanted to share why I went with the IMX462 and plan for it.

So the original plan was to run the IMX462 via INDI server on an RPi Zero 2 W or even a Orange Pi Zero LTS to have wired connectivity, with indi-allsky running remotely indoors on one of my docker hosts.

Haven't got to that full stage yet, and have been testing with an RPi3 running INDI Server, [INDI Web Manger](https://github.com/knro/indiwebmanager), [indi_pylibcamera ](https://github.com/scriptorron/indi_pylibcamera) and the IMX462 outside, and indi-allsky running nicely on docker on a host inside.

This was due to me not fully trusting the weatherproofing/waterproofing of my outdoor enclosure fully, and not wanting to risk ruining the expensive RPi4 and my ASI224MC camera with water damage.

<br>

I found the IMX462 - https://www.arducam.com/product/arducam-2mp-ultra-low-light-starvis-imx462-motorized-ir-cut-camera-for-raspberry-pi/ also had slightly higher resolution than the IMX224 in the ASI224, and was substantially cheaper than the ASI224 and ASI462MC. I have since discovered the Player One Astronomy cameras, and while they are cheaper than the ASIs, they're still not as cheap as the board cameras like the Arducam IMX462.

After a lot of back and forth with Aaron, and I have to say he has been great, and I really appreciate all the help you've given so far Aaron, it looks like currently running it with a remote INDI Server and getting good quality images is going to be a bit harder than I originally thought.

My next step is to have a crack with https://github.com/indilib/indi-3rdparty/tree/master/indi-libcamera and see if the results from that are any better. There's been a patch to libcamera to try and fix some of the IMX462 issues, and it looks easier to get the patch into that than indi_pylibcamera.

I'll report back here when I've had a crack at that and see how I go.

**Comment by Vertikar** [comment]
Hey @aaronwmorris, had a look at your ansible build scripts for INDI server, does it build all 3rd party drivers or just a selection?

**Reply by aaronwmorris** [reply]
By default, it builds all drivers, but I believe if you passed the correct tags, it would build only specific libs/drivers.  I have not really been maintaining the Ansible scripts, in fact I was considering removing them soon.  The shell build script is just easier to maintain.

I will look at making an enhancement to the shell build script to only build specific drivers and libs.

**Comment by Vertikar** [comment]
@aaronwmorris looking to move my IMX462 to my RPi4, but run into an issue, 
i'm getting weird images from it, like this one:
![image](https://github.com/aaronwmorris/indi-allsky/assets/436414/1e7e833d-3414-459e-826b-6252ee6d2d70)

I can't reproduce this on either of my RPi3s. Have made sure they're all on the same kernel and libcamera version.

I remember you tested with your IMX462 in the issue I raised- https://github.com/aaronwmorris/indi-allsky/issues/878#issuecomment-1675642785
was that on a 3 or a 4?

**Reply by aaronwmorris** [reply]
I think I tested on my RPi3.  Did you remember to set the `clock-frequency` on the Rpi4?
